static/js/controllers/confirm-modal.js

```diff
@@ -81,21 +90,13 @@
           if (result.then && typeof result.then === 'function') {
             // It's a promise!
             result
-              .then(function() {
-                $scope.ok();
-              })
-              .catch(function(err) {
-                _setErrorMode(err);
-              });
+              .then($scope.ok)
+              .catch(_setErrorMode);
             return;
           }
 
           // Simple true/false result.
-          if (result) {
-            $scope.ok();
-          } else {
-            _setErrorMode();
-          }
+          $scope.ok();
```

Comment:
That removes some logic : if processing function is synchronous, it returns a boolean, that indicates whether processing worked (saving, typically)
Did you remove it on purpose?